ui/src/components/credentials/SelectSchema.tsx

```diff
@@ -6,11 +6,17 @@ import { getSchemas } from "src/adapters/api/schemas";
 import { useEnvContext } from "src/contexts/env";
 import { Schema } from "src/domain";
 import { ROUTES } from "src/routes";
-import { AsyncTask, isAsyncTaskDataAvailable } from "src/utils/async";
+import { AsyncTask, isAsyncTaskDataAvailable, isAsyncTaskStarting } from "src/utils/async";
 import { isAbortedError, makeRequestAbortable } from "src/utils/browser";
 import { SCHEMA_TYPE } from "src/utils/constants";
 
-export function SelectSchema({ schemaID }: { schemaID: string | undefined }) {
+export function SelectSchema({
+  onSelect,
+  schemaID,
+}: {
+  onSelect: (schema: Schema) => void;
+  schemaID: string | undefined;
```

Comment:
Now that you have onSelect here we can do `fetchSchemas` in the parent because is weird that the info in the parent comes from the children calling the api